src/mode/modeHandler.ts

```diff
@@ -22,23 +23,20 @@ export class ModeHandler implements vscode.Disposable {
         this._modes = [
             new NormalMode(this._motion),
             new InsertMode(this._motion),
+            new VisualMode(this._motion, this),
         ];
 
         this.setCurrentModeByName(ModeName.Normal);
     }
 
     get currentMode() : Mode {
-        let currentMode = this._modes.find((mode, index) => {
-            return mode.isActive;
-        });
-
-        return currentMode;
+        return this._modes.find(mode => mode.isActive);
     }
 
     setCurrentModeByName(modeName : ModeName) {
-        this._modes.forEach(mode => {
+        for (let mode of this._modes) {
```

Comment:
Why? IMO, `for of` is much better than `forEach` - it reads way nicer, and you can do `for (const ... of)` which is not possible with `forEach`. (Are you sure you haven't confused it with `for in`?)